calico/networking/configuring/add-maglev-load-balancing.mdx

```diff
@@ -0,0 +1,34 @@
+---
+description: Add Maglev load balancing to a Kubernetes service.
+---
+
+# Add Maglev load balancing to a service
+
+This guide shows you how to enable Maglev-style load balancing to a service.
+Maglev's consistent-hashing load balancing algorithm minimizes the amount of dropped packets and improves perfomance.
+This is especially useful for maintaining sessions during a load balancer failover.
+
+## Prerequisites
+
+* Your Kubernetes cluster uses $[prodname] for networking.
```

Comment:
Just thought of something else - while I'm conscious of falling down the slippery slope of over-explaining Maglev, I wonder if it's worth noting here that "loadbalancing" nodes and nodes that host the backing pods should be separate. I'm undecided on it. It might become apparent to the user pretty quickly without us telling them. Maybe it goes without saying that we are mitigating loadbalancer downtime, and not pod downtime

Reply:
What does this exactly say? You need calico ebpf, period. Afaict you may use non-calicoy CNI

Reply:
> I wonder if it's worth noting here that "loadbalancing" nodes and nodes that host the backing pods should be separate.

There is no need for them to be separate.

Reply:
> Afaict you may use non-calicoy CNI

Can you give an example of this? 

I thought we needed this prerequisite to exclude Calico-for-policy-only scenarios, such as using AWS VPC CNI with Calico for policy.

Reply:
> There is no need for them to be separate.
If theyre co-located, we don't get any benefit from maglev :thinking: churning a LB will kill backing pods

Reply:
> > There is no need for them to be separate.
> 
> If theyre co-located, we don't get any benefit from maglev 🤔 churning a LB will kill backing pods

yes, some backends will die. I think this could be a good "best practice - do not schedule pods on those nodes because ... " but it is equally important to say that all nodes can act like the LBs. It is up to the user to set their network up for their usecase.